src/evaluators/response/PHAL_ResponseSquaredL2DifferenceSide_Def.hpp

```diff
@@ -40,17 +55,32 @@ ResponseSquaredL2DifferenceSideBase(Teuchos::ParameterList& p, const Teuchos::RC
 
   layout->dimensions(dims);
 
+  std::string sideSetNameForMetric =
+      (plist->isParameter("Is Side Set Planar") && plist->get<bool>("Is Side Set Planar")) ?
```

Comment:
Minor: since plist is not const, you can simply do
```
std::string sideSetNameForMetric = siteSetName +
       (plist->get<bool>("Is Side Set Planar", false) ? "_planar" : "");
```
Might make the line easier to visually parse. (I wish Teuchos::ParameterList had an easy way to provide a default (if key is not found) even for const plists...)

Reply:
I know it's verbose, but I don't like that `get()` creates the parameter in the list. I spent quite some time in the past trying to find where the parameterlists are modified and default options added, when things go wrong.
I'm leaning towards making plist const.